app/models/document.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+class Document < ActiveRecord::Base
+
+  validate :validate_event_type, :on => [:create, :update]
+  validate :event_type_already_exist, :on => :create
+
+  EVENTS = [['-- Select --', 0], ['Calendar of events', 'calendar_of_event']].freeze
+
+  scope :calendar_of_event, -> { where(event_type: 'calendar_of_event') }
+
+  validates :url, :file_name, :presence => true, uniqueness: true
+
+  validate :google_document, :on => [:create, :update]
+
+  def validate_event_type
+    errors.add(:event_type, 'Please select event_type') if event_type.to_s == "0"
+  end
+
+
+  def event_type_already_exist
+    return unless Document.where(event_type: event_type).present?
+
+    errors.add(:event_type, "#{event_type.titleize} type already created. Please use another type.")
+  end
+
+
+  # Call GoogleApiClient to export google document.
+  def google_document
+    # Eg: url = https://docs.google.com/document/d/1iBzIYM_GG5OCXkuF4vKwSYRFaH3gd8Q_kuDrqT7Iu4U/edit
+    # Split google document url than collect document-id.
+    # document_id = 1iBzIYM_GG5OCXkuF4vKwSYRFaH3gd8Q_kuDrqT7Iu4U
+    document_id = URI.split(url)[5].split('/')[3]
+    GoogleApiClient.export_file(document_id, "application/pdf")
+  rescue StandardError => error
+    errors.add(:url, google_client_error_message(error))
+  end
+
+
+  def self.calendar_event
```

Comment:
It could be confusing for future developers to have both `calendar_event` and `calendar_of_event` — they sound the same. Maybe we could just have the scope `calendar_of_event` just return the `first` and then get rid of this method. Would that work?